plugins/inputs/nfsstat/nfsstat.go

```diff
@@ -0,0 +1,289 @@
+		"seek",
+		"allocate",
+		"deallocate",
+		"layoutstats",
+		"clone",
+		"copy",
+		"offload_cancel",
+		"lookupp",
+		"layouterror",
+		"copy_notify",
+		"getxattr",
+		"setxattr",
+		"listxattrs",
+		"removexattr",
+	}
+}
+
+func getNFSOp(index int, nfsvers int) (string, error) {
+
+	var nfsops []string
+	switch nfsvers {
+	case 3:
+		nfsops = getNFSv3Ops()
+	case 4:
+		nfsops = getNFSv4Ops()
+	default:
+		return "", ErrorInvalidNFSVersion
+	}
+
+	// The operation index is defined by the slice with the operation
+	// names. Catch potential programming errors if index is out of bounds
+	if index < 0 || index >= len(nfsops) {
+		return "", ErrorInvalidNFSOpIndex
+	}
+
+	return nfsops[index], nil
+}
+
+func parseCounters(s string) ([]int64, error) {
+	ss := strings.Split(s, " ")
+	sn := make([]int64, len(ss))
+	for i, val := range ss {
+		val_i, err := strconv.ParseInt(val, 0, 64)
+		if err != nil {
+			return []int64{}, err
+		}
+		if val_i < 0 {
+			//This should never happen. Lines come from the kernel
+			// and must not have negative or invalid numbers.
+			return []int64{}, ErrorInvalidCounterValue
+		}
+		sn[i] = val_i
+	}
+	return sn, nil
+}
+
+func parseStatLine(line string, nfsvers int) ([]int64, error) {
+
+	var s string
+
+	switch nfsvers {
+	case 3:
+		s = line[len(prefix_nfs3):]
+	case 4:
+		s = line[len(prefix_nfs4):]
+	default:
+		return []int64{}, ErrorInvalidNFSVersion
+	}
+
+	stats, err := parseCounters(s)
+	if err != nil {
+		return []int64{}, err
+	}
+	return stats, nil
+}
+
+func processNFSstatLine(line string, nfsvers int) ([]NFSStatCounter, error) {
+
+	var counters []NFSStatCounter
+	nfsstat, err := parseStatLine(line, nfsvers)
+	if err != nil {
+		return []NFSStatCounter{}, err
+	}
+
+	for i, val := range nfsstat {
+		op, err := getNFSOp(i, nfsvers)
+		if err != nil {
+			return []NFSStatCounter{}, err
+		}
+		counters = append(counters,
+			NFSStatCounter{
+				nfsvers: strconv.Itoa(nfsvers),
+				op:      op,
+				val:     val,
+			})
+	}
+
+	return counters, nil
+}
+
+func processNFSStats(contents []string) ([]NFSStatCounter, error) {
+
+	var counters []NFSStatCounter
+
+	for _, line := range contents {
+		nfsvers, isnfsstat := getNFSVers(line)
+		if !isnfsstat {
+			continue
+		}
+		lcounters, err := processNFSstatLine(line, nfsvers)
+		if err != nil {
+			return []NFSStatCounter{}, err
+		}
+		counters = append(counters, lcounters...)
+	}
+	return counters, nil
+}
+
+func assignStats(counters []NFSStatCounter, acc telegraf.Accumulator) error {
+
+	var fields = make(map[string]interface{})
+	for _, c := range counters {
+		tags := map[string]string{
+			"nfsvers": c.nfsvers,
+			"op":      c.op,
+		}
+		fields["operations"] = c.val
+		acc.AddFields("nfsstat", fields, tags)
+	}
+
+	return nil
+}
+
+func (n *NFSStat) Gather(acc telegraf.Accumulator) error {
+
+	rawstats, err := ioutil.ReadFile("/proc/net/rpc/nfs")
+	if err != nil {
+		n.Log.Errorf("Failed to read /proc/net/rpc/nfs:  ", err)
+		return err
```

Comment:
don't need to log as well as return. if you want the extra context use 
```suggestion
		return fmt.Errorf("reading /proc/net/rpc/nfs: %w", err)
```